src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureUsageProcessor.kt

```diff
@@ -0,0 +1,141 @@
+        usageInfo: UsageInfo?,
+        beforeMethodChange: Boolean,
+        usages: Array<out UsageInfo>?
+    ): Boolean {
+        if (beforeMethodChange) return false
+
+        val rsChangeInfo = changeInfo as? RsSignatureChangeInfo ?: return false
+        val usage = usageInfo as? RsFunctionUsage ?: return false
+        val config = rsChangeInfo.config
+        if (usage is RsFunctionUsage.MethodImplementation) {
+            processFunction(config.function.project, config, usage.overriddenMethod)
+        }
+        else {
+            processFunctionUsage(config, usage)
+        }
+
+        return false
+    }
+
+    override fun processPrimaryMethod(changeInfo: ChangeInfo?): Boolean {
+        val rsChangeInfo = changeInfo as? RsSignatureChangeInfo ?: return false
+        val config = rsChangeInfo.config
+        val function = config.function
+        val project = function.project
+
+        processFunction(project, config, function)
+        return false
+    }
+
+    override fun shouldPreviewUsages(changeInfo: ChangeInfo?, usages: Array<out UsageInfo>?): Boolean = false
+
+    override fun setupDefaultValues(
+        changeInfo: ChangeInfo?,
+        refUsages: Ref<Array<UsageInfo>>?,
+        project: Project?
+    ): Boolean {
+        return true
+    }
+
+    override fun registerConflictResolvers(
+        snapshots: MutableList<ResolveSnapshotProvider.ResolveSnapshot>?,
+        resolveSnapshotProvider: ResolveSnapshotProvider,
+        usages: Array<out UsageInfo>?,
+        changeInfo: ChangeInfo?
+    ) {
+    }
+}
+
+private fun findVisibilityConflicts(
+    function: RsFunction,
+    config: RsChangeFunctionSignatureConfig,
+    usages: Array<UsageInfo>,
+    map: @NotNull MultiMap<PsiElement, String>
+) {
+    val functionUsages = usages.filterIsInstance<RsFunctionUsage>()
+    val clone = function.copy() as RsFunction
+    changeVisibility(clone, config)
+
+    for (usage in functionUsages) {
+        val sourceModule = (usage.element as? RsElement)?.containingMod ?: continue
+        if (!clone.isVisibleFrom(sourceModule)) {
+            val moduleName = sourceModule.qualifiedName.orEmpty()
+            map.putValue(function, RsBundle.message("refactoring.change.signature.visibility.conflict", moduleName))
+        }
+    }
+}
+
+private fun findNameConflicts(
+    function: RsFunction,
+    config: RsChangeFunctionSignatureConfig,
+    map: @NotNull MultiMap<PsiElement, String>
+) {
+    val (owner, items) = when (val owner = function.owner) {
+        is RsAbstractableOwner.Impl -> owner.impl to owner.impl.expandedMembers
+        is RsAbstractableOwner.Trait -> owner.trait to owner.trait.expandedMembers
+        else -> {
+            val parent = function.parent as RsItemsOwner
+            parent to parent.itemsAndMacros.filterIsInstance<RsItemElement>().toList()
+        }
+    }
+    for (item in items) {
+        if (item == function) continue
+        if (item.name == config.name) {
```

Comment:
Btw there will be a false-positive in case of multiple functions with same name but different cfg-attributes:
```rust
#[cfg(unix)]
fn foo() {}
#[cfg(windows)]
fn foo() {}
```
But I think it can be ignored for now